csrc/executor.h

```diff
@@ -68,6 +68,13 @@ class FusionExecutor : public NonCopyable {
       Fusion* fusion,
       const KernelArgumentHolder& args);
 
+  //! Only a single PrecomputedValues is active for a fusion. Consequently, it
+  //! can be necessary to pass PrecomputedValues within a given scope.
+  KernelArgumentHolder inferOutputSizes(
+      Fusion* fusion,
+      const KernelArgumentHolder& args,
+      PrecomputedValues* evaluator_precomputed_values);
```

Comment:
Cleanup suggestion: we can probably combine this and the flavor before by allowing this parameter to be nullptr.